storage/database.go

```diff
@@ -100,7 +99,7 @@ type db struct {
 
 	// Contains an entry to all shards for fast shard lookup, an
 	// entry will be nil when this shard does not belong to current database
-	shards []databaseShard
+	shards map[uint32]databaseShard
```

Comment:
but since the shards here are no longer consecutive, we will have to loop through the array to look up a particular shard.